pkg/collector/corechecks/cluster/orchestrator/processors/k8s_cluster.go

```diff
@@ -0,0 +1,161 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver && orchestrator
+// +build kubeapiserver,orchestrator
+
+package processors
+
+import (
+	"context"
+	"fmt"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+	"github.com/DataDog/datadog-agent/pkg/orchestrator"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/twmb/murmur3"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	corev1Client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// K8sClusterProcessor is a processor for Kubernetes clusters. There is no
+// concept of cluster per se in Kubernetes. The model is created by aggregating
+// data from the Kubernetes Node resource and pulling API Server information.
+// This is why that processor is custom and not following the generic logic like
+// other resources.
+type K8sClusterProcessor struct {
+	Processor
+	nodeHandlers Handlers
+}
+
+// NewK8sClusterProcessor creates a new processor for Kubernetes clusters.
+func NewK8sClusterProcessor() *K8sClusterProcessor {
+	return &K8sClusterProcessor{
+		nodeHandlers: new(K8sNodeHandlers),
+	}
+}
+
+// Process is used to process a list of node resources forming a cluster.
+func (p *K8sClusterProcessor) Process(ctx *ProcessorContext, list interface{}) (messages []model.MessageBody, processed int, err error) {
```

Comment:
Seeing this I can tell throughout the PR that you prefer named returns in terms of interface documentation. 

I don't have a big preference here regarding the usage, but we should discuss on this coding style whether or not to use them in those bigger chunks here (might be totally fine)

Reply:
Although the 3+ parameters from https://github.com/DataDog/datadog-agent/pull/10274/files#r790808566 would also apply here the main reason for that method is elsewhere: panic detection. This allows us to do a little trick: by setting the return arg to a value at the very beginning like this, we're sure that if the code panics we'll get that  value and not the type's default value. This way we can detect if a panic occurred while a zero value would just mean that we've processed no resource for other reasons.

Reply:
yes I agree, the main reason I am afraid of named returns is accidental shadowing and not catching it during a review. 
But let's decide this as a team what we prefer, I do see the advantages (documentation, forced behaviour/returns)